src/unstract/sdk/file_storage/impl.py

```diff
@@ -218,17 +196,13 @@ def modification_time(self, path: str) -> datetime:
         Returns:
             datetime: Last modified time in datetime
         """
-        try:
-            file_info = self.fs.info(path)
-            file_mtime = file_info["mtime"]
-            if not isinstance(file_mtime, datetime):
-                file_mtime = datetime.fromtimestamp(file_mtime)
-            return file_mtime
-        except FileNotFoundError as e:
-            raise e
-        except Exception as e:
-            raise FileOperationError(str(e)) from e
+        file_info = self.fs.info(path)
+        file_mtime = file_info["mtime"]
```

Comment:
Don't know the full context.
`.get` might be a better choice here. If `mtime` is not present in `fileinfo`, using direct access could raise a KeyError

Reply:
Same as above